keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1800 @@
+that might indicate a serious problem?
+-->
+
+A significant increase in the following metrics after enabling the feature would
+be a strong indicator that a rollback is needed:
+
+-   `topology_manager_admission_errors_total`: Indicates that the Topology
+    Manager is rejecting pods that it would have previously admitted.
+-   `cpu_manager_pinning_errors_total`: Indicates failures in the CPU allocation
+    logic.
+-   `memory_manager_pinning_errors_total`: Indicates failures in the Memory
+    allocation logic.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+Manual testing of the upgrade/downgrade path will be performed as part of the
+alpha development cycle.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+N/A.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+An operator can determine if the feature is in use by inspecting the Kubelet
+configuration on each node to see if the `PodLevelResourceManagers` feature gate
+is enabled and if the `--topology-manager-scope` is set to `pod` or `container`.
+Additionally, inspecting the logs for messages related to pod-level allocation
+would indicate usage.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+
+- [ ] Events
+  - Event Reason:
+- [ ] API .status
+  - Condition name:
+  - Other field:
+- [ ] Other (treat as last resort)
+  - Details:
+
+-->
+
+-   [X] Events
+    -   Event Reason: A pod failing admission due to a `TopologyAffinityError`
+        will generate an event on the pod, visible via `kubectl describe pod`.
+-   [X] Other (treat as last resort)
+    -   Details: An operator can check the `cpu_manager_pinning_requests_total`
```

Comment:
maybe add _pod_ pinning metric? or variants of the above?

Reply:
I updated this info with the new proposed metrics.